### ISSUE TITLE
RCS1118 False positive when variable used in `ref` extension method

### ISSUE THREAD

**Issue by eduherminio** [JosefPihrt/Roslynator#947]
**Product and Version Used**: VS & Roslynator 2022 v4.1.1

**Steps to Reproduce**:

```csharp
public static class BoardExtensions
{
    public static ulong Foo(int square)
    {
        ulong board = default; // RCS1118 -> Mark as const
        board.Bar(square);

        return board;
    }

    public static ulong Bar(this ref ulong board, int square)
    {
        return board |= (1UL << square);
    }
}
```

**Actual Behavior**:
RCS1118 and suggestion to make const

**Expected Behavior**:
No RCS1118, since making it const breaks compilation since the variable is actually modified.
